src/wcontrol/Foreman.cc

```diff
@@ -26,49 +26,73 @@
 
 // System headers
 #include <cassert>
-#include <deque>
-#include <iostream>
-#include <memory>
-#include <mutex>
 #include <thread>
-#include <utility>
+
+// Third party headers
+#include "boost/filesystem.hpp"
 
 // LSST headers
 #include "lsst/log/Log.h"
 
 // Qserv headers
-#include "global/UnsupportedError.h"
 #include "mysql/MySqlConfig.h"
-#include "proto/worker.pb.h"
-#include "wbase/Base.h"
-#include "wbase/SendChannelShared.h"
+#include "qhttp/Request.h"
+#include "qhttp/Response.h"
+#include "qhttp/Server.h"
 #include "wbase/WorkerCommand.h"
+#include "wconfig/WorkerConfig.h"
 #include "wcontrol/SqlConnMgr.h"
-#include "wcontrol/TransmitMgr.h"
 #include "wcontrol/WorkerStats.h"
 #include "wdb/ChunkResource.h"
-#include "wdb/QueryRunner.h"
+#include "wdb/SQLBackend.h"
+#include "wpublish/QueriesAndChunks.h"
+
+using namespace std;
+namespace fs = boost::filesystem;
+namespace qhttp = lsst::qserv::qhttp;
 
 namespace {
 LOG_LOGGER _log = LOG_GET("lsst.qserv.wcontrol.Foreman");
-}
 
-using namespace std;
+/// Remove the result file specified in the parameter of the method.
+/// @param fileName An absolute path name to a file to be removed.
+/// @return The HTTP status (200 - success, 404 - file doesn't exist, 500 - operation failed).
+int removeResultFile(std::string const& fileName) {
+    string const context = "Foreman::" + string(__func__) + " ";
+    fs::path const filePath(fileName);
+    if (!fs::exists(filePath)) return 404;
+    boost::system::error_code ec;
+    fs::remove_all(filePath, ec);
+    if (ec.value() != 0) {
+        LOGS(_log, LOG_LVL_WARN,
+             context << "failed to remove the result file: " << fileName << ", code: " << ec.value()
+                     << ", error:" << ec.message());
+        return 500;
+    }
+    LOGS(_log, LOG_LVL_DEBUG, context << "result file removed: " << fileName);
+    return 200;
```

Comment:
Error codes `200`, `404`, `500` is the "Lingua franca" of any programmer who deals with HTTP/HTTPS protocols. These constants are more recognizable in this context than any symbolic definition. If we were to introduce such definitions in `qhttp` rather than here. If you look at the implementation file for the class `qhttp::Response.cc` then you can see all known error codes defined by the HTTP standard:
```
std::map<unsigned int, const std::string> responseStringsByCode = {
        ...
        {200, "OK"},
        ...
        {500, "Internal Server Error"},
        ...
};
```
I will ask @fritzm if adding a new header with symbolic definitions of these codes to `qhttp` would be a good idea from his point of view.

Reply:
The status coded have been added to `qhttp` in https://jira.lsstcorp.org/browse/DM-39907. I'm going to migrate `wcontrol::Foreman` to use the symbolic definitions of the corresponding codes rather than the "magic" constants. Thank you for the suggestion!